plugin/trino-pinot/src/test/java/io/trino/plugin/pinot/TestDynamicTable.java

```diff
@@ -448,4 +448,28 @@ public void testQuotesInColumnName()
         assertEquals(extractPql(dynamicTable, TupleDomain.all()), expectedPql);
         assertEquals(dynamicTable.getTableName(), tableName);
     }
+
+    @Test
+    public void testQueryOptions()
+    {
+        String tableName = realtimeOnlyTable.getTableName();
+        String tableNameWithSuffix = tableName + REALTIME_SUFFIX;
+        String query = """
+                SET skipUpsert=true;
+                SET useMultistageEngine=true;
+                SELECT FlightNum
+                FROM %s
+                LIMIT 50;
+                """.formatted(tableNameWithSuffix);
+        DynamicTable dynamicTable = buildFromPql(pinotMetadata, new SchemaTableName("default", query), mockClusterInfoFetcher, TESTING_TYPE_CONVERTER);
+        String expectedPql = """
+                SELECT "FlightNum" \
+                FROM %s \
+                LIMIT 50;
+                SET useMultistageEngine = true;
+                SET skipUpsert = true;
```

Comment:
Does moving the `set` statements only enable those options after the query is executed?

Reply:
In one multi-line query statement, SET statements will be executed first on matter the ordering.

Reply:
I would recommend moving SET to first in DynamicTablePqlExtractor or leave a comment in this test. 2/2 reviewers asked the same question. It means the current logic is confusing.

Reply:
done.